### PR TITLE
Remove wchartype dependency and is_multi_byte [1/2]

### PR DESCRIPTION
**Pull request by nsoranzo**
[wchartype](http://ginstrom.com/code/wchartype.html) won't make sense in Python3, where all strings are UTF-8 encoded.

See individual commits for details.

xref. #1715

### CODE REVIEW

**Comment by jmchilton** [comment]
Does ``get_file_peek`` still need to take that is multi-byte argument and just ignore it because of tool shed datatype definitions?

File: lib/galaxy/datatypes/sniff.py [hunk 1]
```diff
@@ -266,7 +259,7 @@ def is_column_based(fname, sep='\t', skip=0, is_multi_byte=False):
     >>> is_column_based(fname)
     True
     """
-    headers = get_headers(fname, sep, is_multi_byte=is_multi_byte)
+    headers = get_headers(fname, sep)
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/sniff.py]: Need to add a try/except UnicodeDecodeError around this line.

**Comment by nsoranzo** [comment]
> Does `get_file_peek` still need to take that is multi-byte argument and just ignore it because of tool shed datatype definitions?

@jmchilton Then I should probably also re-add the `is_multi_byte` param to all the `set_peek()` methods, since some TS datatype subclass may be calling the parent's method when overriding it. What do you think?

**Comment by jmchilton** [comment]
@nsoranzo Sounds fair - sorry about these data types - they are why we can't have nice things.

**Comment by nsoranzo** [comment]
@jmchilton Restored the `is_multi_byte` in the signatures.

**Comment by jmchilton** [comment]
Thanks for supporting those datatypes and all of this - really nice work!

**Comment by nsoranzo** [comment]
Thanks for the review and merge, part 2/2 coming in the next few days, I just have to split changes in commits and test.

**Pull request merged**